lib/shared/inference/anthropic_completions.rb

```diff
@@ -104,17 +105,22 @@ def self.perform!(
                   end
               rescue IOError
                 raise if !cancelled
-              ensure
-                log.update!(
-                  raw_response_payload: response_raw,
-                  request_tokens: DiscourseAi::Tokenizer::AnthropicTokenizer.size(prompt),
-                  response_tokens: DiscourseAi::Tokenizer::AnthropicTokenizer.size(response_data),
-                )
               end
             end
 
             return response_data
           end
+        ensure
+          if block_given?
+            log.update!(
+              raw_response_payload: response_raw,
+              request_tokens: DiscourseAi::Tokenizer::AnthropicTokenizer.size(prompt),
+              response_tokens: DiscourseAi::Tokenizer::AnthropicTokenizer.size(response_data),
+            )
+          end
+          if Rails.env == "development" && log
```

Comment:
```suggestion
          if Rails.env.development? && log
```

Reply:
yeah people like this incantation, but I really dislike it for some reason

Reply:
ok I dug into it... and yeah it is unconditionally better cause it is faster :open_mouth: 

https://github.com//blob/bb52c41dbfdf8d57b58f82319506446b70bca1b1/activesupport/lib/active_support/environment_inquirer.rb#L27-L33